### PR TITLE
[IOTDB-706] Introduce virtual memtable for larger Chunk

### PR DESCRIPTION
**Pull request by qiaojialin**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java [hunk 1]
```diff
@@ -540,6 +574,80 @@ private void releaseFlushedMemTable(IMemTable memTable) {
     }
   }
 
+  /**
+   * Construct the a new or get an existing TsFileSequenceReader of a TsFile.
+   *
+   * @return a TsFileSequenceReader
+   */
+  private TsFileSequenceReader getFileReader(TsFileResource tsFileResource) throws IOException {
```
Comment (samperson1997) [inline server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java:582]: Will this unused method be used later?

File: server/src/main/java/org/apache/iotdb/db/writelog/recover/TsFileRecoverPerformer.java [hunk 2]
```diff
@@ -80,17 +104,33 @@ public TsFileRecoverPerformer(String logNodePrefix, VersionController versionCon
    * @return a RestorableTsFileIOWriter if the file is not closed before crush, so this writer can
```
[the remark this answers is not in the document]
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/writelog/recover/TsFileRecoverPerformer.java]: fixed~

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java (hunk 1 above)
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java]: fixed

File: server/src/main/java/org/apache/iotdb/db/engine/flush/VmMergeUtils.java [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+  private static final Logger logger = LoggerFactory.getLogger(VmMergeUtils.class);
+
+  private VmMergeUtils() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  public static void fullMerge(RestorableTsFileIOWriter writer,
+      List<List<RestorableTsFileIOWriter>> vmWriters, String storageGroup, VmLogger vmLogger,
+      Set<String> devices, boolean sequence) throws IOException {
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/engine/flush/VmMergeUtils.java:57]: This method has a lot of duplicate lines and looks too complicated, it's better to do a refactor or split it to some small methods.

File: server/src/main/java/org/apache/iotdb/db/engine/flush/VmMergeUtils.java [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+          measurementSchemaMap.computeIfAbsent(measurementEntry.getKey(), k ->
+              new MeasurementSchema(k, measurementEntry.getValue().get(0).getDataType()));
+        }
+      }
+    }
+  }
+
+  public static void levelMerge(RestorableTsFileIOWriter writer,
+      List<RestorableTsFileIOWriter> vmWriters, String storageGroup, VmLogger vmLogger,
+      Set<String> devices, boolean sequence) throws IOException {
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/engine/flush/VmMergeUtils.java:219]: Same as `fullMerge`

**Pull request merged**